ship: refactor teams to match reorg (#7277)
sourcegraph/handbook: content/departments/engineering/dev/process/engineering_ownership.md

- sourcegraph/learn
- sourcegraph/docsite

> ## [Delivery](../../teams/release/index.md) (and Delivery Tiger Team)

Reviewer: Should the header change to `Release`?

> ## [Delivery](../../teams/release/index.md) (and Delivery Tiger Team)

Reviewer 2: ```suggestion
## [Delivery](../../teams/release/index.md)
```
Removing references to the Delivery Tiger Team

- [sourcegraph/server](https://github.com/sourcegraph/sourcegraph/tree/main/cmd/server)
- [sourcegraph/worker](https://github.com/sourcegraph/sourcegraph/tree/main/cmd/worker)

> ## [Developer Experience](../../teams/devinfra/index.md)

Reviewer: Should the header change to `DevInfra`?

Reviewer 3: ```suggestion
## [Developer Infrastructure](../../teams/devinfra/index.md)
```

- It is preferred to list a team (e.g., @sourcegraph/frontend-platform) rather than individuals.
- If there is an individual with expertise in a given part of the code, who is not on the owning team (e.g., Thorsten Ball with sg), that person should be listed in OWNERS.

> For questions about our use of owners files, please reach out on #devinfra.

Reviewer: ```suggestion
For questions about our use of owners files, please reach out on #discuss-dev-experience.
```

Note that `#ask-dev-infra` actually doesn't exist, it's still called `#ask-dev-experience`
